### PR TITLE
C2s/ping

### PR DESCRIPTION
**Pull request by NelsonVides**

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/134446) / small_tests / 48cf8eb61472c79b3bb61baf3b86c662f2b759d7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3748/134446/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3748/134446/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-02_22.04.29/lib.mongooseim.logs/run.2022-09-02_22.04.29/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/134641) / small_tests / ddfcc46594905daae9e076ad286f4bb4daafe209
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3748/134641/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3748/134641/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-03_14.23.44/lib.mongooseim.logs/run.2022-09-03_14.23.45/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/134798) / small_tests / b0e3f1d12bcf0030d4fc69d9b031077666cbac25
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3748/134798/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3748/134798/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-03_14.47.07/lib.mongooseim.logs/run.2022-09-03_14.47.07/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/137998) / small_tests / a729fc488171b6d25ff84f3047491b342f091b07
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3748/137998/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3748/137998/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-20_14.21.22/lib.mongooseim.logs/run.2022-09-20_14.21.22/suite.log.html)

**Comment by mongoose-im** [comment]
[ldap_mnesia_24](https://circleci.com/gh/esl/MongooseIM/138922) / ldap_mnesia / c526baf29fe1761f62a339cc45732bc7220000ee
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3748/138922/ldap_mnesia.24.3.4)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3748/138922/ldap_mnesia.24.3.4/big/ct_run.test%4003244095980d.2022-09-23_13.32.35/index.html)
**OK: 829** / Failed: 0 / **User-skipped: 427** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/141013) / small_tests / 002eb936d0f39d13664c7c7b018b27bd3271b40a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3748/141013/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3748/141013/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-03_20.49.51/lib.mongooseim.logs/run.2022-10-03_20.49.52/suite.log.html)

File: src/mod_ping.erl [hunk 1]
```diff
@@ -152,30 +122,60 @@ iq_ping(Acc, _From, _To, #iq{sub_el = SubEl} = IQ, _) ->
...
+                {Id, Id} ->
+                    Jid = mongoose_c2s:get_jid(StateData),
+                    TDelta = erlang:monotonic_time(millisecond) - T0,
+                    mongoose_hooks:user_ping_response(HostType, #{}, Jid, IqResponse, TDelta),
+                    Action = {{timeout, ping_timeout}, cancel},
+                    {stop, mongoose_c2s_acc:to_acc(Acc, actions, Action)};
+                _ ->
+                    {ok, Acc}
+            end;
+        {_, {error, not_found}} ->
```
Comment (kamilwaz) [inline src/mod_ping.erl:143]: This clause is redundant as a next one returns the same result.

**Review by kamilwaz — CHANGES_REQUESTED** [review]
Looks good to me, I just have one small comment.

File: src/mod_ping.erl (hunk 1 above)
Comment (NelsonVides) [reply src/mod_ping.erl:143]: Indeed 👌🏽

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/141473) / small_tests / a73bc335c9606220ad40f9e8d7b61545df816ddc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3748/141473/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3748/141473/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-04_11.14.00/lib.mongooseim.logs/run.2022-10-04_11.14.00/suite.log.html)

**Review by kamilwaz — APPROVED** [review]
Excellent 👍

**Pull request merged**